Source/WebCore/style/AnchorPositionEvaluator.cpp

```diff
@@ -267,6 +474,19 @@ LayoutRect AnchorPositionEvaluator::computeAnchorRectRelativeToContainingBlock(C
     auto anchorHeight = anchorBox->offsetHeight();
     auto anchorLocation = LayoutPoint { offsetFromAncestorContainer(anchorBox, containingBlock) };
 
+    if (&containingBlock == &containingBlock.view() && anchoredBox.isFixedPositioned()) {
+        // Handle fixed positioning x scrolling anchor.
+        bool isFixedAnchor = false;
+        for (const RenderElement* box = anchorBox.ptr(); box && box != &containingBlock; box = box->container()) {
+            if (box->isFixedPositioned()) {
```

Comment:
Note that this only consults style, and not whether some container (e.g. a transform or filter) prevents fixedpos behavior). For that you'd consult `zlayer->behavesAsFixed()`.